main/src/ca/uwaterloo/flix/api/lsp/provider/completion/Priority.scala

```diff
@@ -16,79 +16,52 @@
 package ca.uwaterloo.flix.api.lsp.provider.completion
 
 /**
-  * Priority for completions.
-  *
-  * Using a priority, we can make certain suggestions occur earlier in
-  * the list of suggestions by generating a SortText using `Priority.toSortText`
-  * on the relevant priority and label. This SortText can then be used in the `CompletionItem`
-  * for the suggestion.
-  *
-  * In practice, we leverage the alphabetical sorting of completions by
-  * associating each priority with a number. The higher the priority,
-  * the lower the number. Then `Priority.toSortText`
-  * prepends that number to the label. This results in completions which
-  * are firstly sorted by the priority and then by name.
+  * Represents the priority of a completion.
   *
+  * A completion consists of an (A) absolute priority (e.g., High or Low) and (B) a relative priority (e.g., High(5)).
   */
 sealed trait Priority {
-
-  /**
-    * Returns a priority that is one lower than `this`.
-    */
-  def downgrade: Priority = this match {
-    case Priority.Highest => Priority.Higher
-    case Priority.Higher => Priority.High
-    case Priority.High => Priority.MediumHigh
-    case Priority.MediumHigh => Priority.Medium
-    case Priority.Medium => Priority.MediumLow
-    case Priority.MediumLow => Priority.Low
-    case Priority.Low => Priority.Lower
-    case Priority.Lower => Priority.Lowest
-    case Priority.Lowest => Priority.Lowest
-  }
-
+  /** Returns the relative priority of `this` priority. */
+  def relative: Int
```

Comment:
Are relative numbers allowed to be negative? I think either way it would be good to note

Reply:
Fixed and documented.